src/main/java/com.gigssandbox/UserActivity.java

```diff
@@ -12,7 +12,7 @@
 import com.gigssandbox.services.GigService;
 import com.gigssandbox.services.UserService;
 
-class UserActivity {
+class UserActivity implements Runnable {
```

Comment:
that is not the best solution. its better to create anonymous class in UserActivityManager